SU2_CFD/src/output/CFlowOutput.cpp

```diff
@@ -3277,24 +3277,33 @@ void CFlowOutput::WriteForcesBreakdown(const CConfig* config, const CSolver* flo
   // clang-format on
 }
 
-bool CFlowOutput::WriteVolume_Output(CConfig *config, unsigned long Iter, bool force_writing){
+bool CFlowOutput::WriteVolume_Output(CConfig *config, unsigned long Iter, bool force_writing, unsigned short iFile){
+  
+  bool writeRestart = false;
+  unsigned long *VolumeFrequencies = config->GetVolumeOutputFrequencies();
+  auto FileFormat = config->GetVolumeOutputFiles();
 
   if (config->GetTime_Domain()){
     if (((config->GetTime_Marching() == TIME_MARCHING::DT_STEPPING_1ST) || (config->GetTime_Marching() == TIME_MARCHING::TIME_STEPPING)) &&
-        ((Iter == 0) || (Iter % config->GetVolume_Wrt_Freq() == 0))){
+        ((Iter == 0) || (Iter % VolumeFrequencies[iFile] == 0))){
       return true;
     }
 
+    /* check if we want to write a restart file*/ 
+    if (FileFormat[iFile] == OUTPUT_TYPE::RESTART_ASCII || FileFormat[iFile] == OUTPUT_TYPE::RESTART_BINARY || FileFormat[iFile] == OUTPUT_TYPE::CSV) {
+      writeRestart = true;  
+    }
+
+    /* only write 'double' files for the restart files */
     if ((config->GetTime_Marching() == TIME_MARCHING::DT_STEPPING_2ND) &&
-        ((Iter == 0) ||
-         (Iter % config->GetVolume_Wrt_Freq() == 0) ||
-         ((Iter+1) % config->GetVolume_Wrt_Freq() == 0) || // Restarts need 2 old solution.
-         ((Iter+2) == config->GetnTime_Iter()))){ // The last timestep is written anyways but again one needs the step before for restarts.
+      ((Iter == 0) || (Iter % VolumeFrequencies[iFile] == 0) ||
+      (((Iter+1) % VolumeFrequencies[iFile] == 0) && writeRestart==true) || // Restarts need 2 old solutions.
+      (((Iter+2) == config->GetnTime_Iter()) && writeRestart==true))){      // The last timestep is written anyway but one needs the step before for restarts.
```

Comment:
Btw this logic does not work when an unsteady simulation stops by reaching the maximum time.

Reply:
I don't see it. If I set 
TIME_ITER=10 
OUTPUT_FILES= (RESTART, PARAVIEW_MULTIBLOCK)
OUTPUT_WRT_FREQ= 6,6

I get restarts at 0, {5,6} and at {8,9}

Reply:
I was referring to `MAX_TIME`, if that is lower than `TIME_ITER * TIME_STEP` the simulation stops before `TIME_ITER` and the "n-1" file will not be outputted, one alternative would be to modify TIME_ITER internally such that `TIME_ITER * TIME_STEP = MAX_TIME` but that is not as straight forward as it sounds...
I was just pointing it out, does not need to be fixed here.